src/title.hpp

```diff
@@ -0,0 +1,179 @@
+#pragma once
+
+#include <string>
+#include <iostream>
+
+#include "utils.hpp"
+
+
+//concatenates all strings in vector into a single string
+std::string multiple_lines_into_one(std::vector<std::string>& lines) {
+    std::string title = lines[0];
+    for (size_t i = 1; i < lines.size(); i++) {
+        append_line(title, lines[i]);
+    }
+    return title;
+}
+
+//concatenates all strings in vector into a single string starting from the last one
+std::string multiple_lines_into_one_reverse(std::vector<std::string>& lines) {
+    std::string title = lines[lines.size() - 1];
+    for (int j = static_cast<int>(lines.size() - 2); j >= 0; j--) {
```

Comment:
Assuming `lines` is not empty is dangerous. It is better to perform checks in the function, instead of assuming that the function is correctly used.
```suggestion
    std::string title;
    for (int j = static_cast<int>(lines.size() - 1); j >= 0; j--) {
```